Dockerfile

```diff
@@ -3,22 +3,29 @@ FROM docker.io/library/centos:8.2.2004
 LABEL maintainer="Matthew F Leader <mleader@redhat.com>"
 
 ENV PATH=/root/.local/bin:$PATH \
-    PIP_NO_CACHE_DIR=off \
     LANG=C.UTF-8
 ENV APP_ROOT=/data_server
-ENV PIPENV_PYTHON=python
+ENV PYTHONPATH=${APP_ROOT}/app
 
-RUN dnf install -y python38 \
+COPY ./app ${APP_ROOT}/app/
+COPY requirements.txt ${APP_ROOT}/
+
+WORKDIR ${APP_ROOT}
+
+# need psycopg2 for sqlalchemy to create db relations
+# psycopg2 dependencies:
+#   gcc, libpq-devel
+
+RUN dnf install -y \
+        gcc \
+        libpq-devel \
+        python38-devel \
+        python38 \
     && ln -s /usr/bin/python3 /usr/bin/python \
     && ln -s /usr/bin/pip3 /usr/bin/pip \
     && dnf clean all \
-    && pip install pipenv
-
-COPY ./app ${APP_ROOT}/app
-COPY Pipfile ${APP_ROOT}/
-COPY ./start.bash ${APP_ROOT}/start.bash
-WORKDIR ${APP_ROOT}
+    && pip install --requirement requirements.txt
```

Comment:
Completely not important but why run this with && instead of another RUN line? More curiosity than issue.

Reply:
It's mostly a habit learned from [containerfile best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run). I know it creates only one layer in this case, instead of a new layer for each `RUN`, but I am having hard time connecting fewer intermediate layers with shorter build times.